openssl-sys/src/ossl_typ.rs

```diff
@@ -1010,7 +1010,41 @@ cfg_if! {
     }
 }
 
-pub enum COMP_METHOD {}
+pub enum COMP_CTX {}
+
+cfg_if! {
+    if #[cfg(ossl110)] {
+        pub enum COMP_METHOD {}
+    } else {
+        #[repr(C)]
+        pub struct COMP_METHOD {
+            pub(crate) type_: c_int,
```

Comment:
Why pub(crate) here? IIRC nothing in openssl-sys would ever be looking at this field in particular.

Reply:
Oh the shim for `COMP_get_type` uses that, but maybe shims should be in consumers of `openssl-sys` rather than `openssl-sys` itself?

Reply:
Yeah, unless there's a `COMP_get_type` macro defined in the OpenSSL sources, I don't think we'd want to make a shim in openssl-sys.

It's fine to just make all fields of openssl-sys structs public unless there's some specific field or type that you don't want to bother properly defining.